tools/clang/include/clang/SPIRV/SPIRVContext.h

```diff
@@ -198,7 +200,12 @@ class SpirvContext {
 
   const StructType *getByteAddressBufferType(bool isWritable);
 
+  SpirvConstant *getConstantUint32(uint32_t value, SourceLocation loc = {});
+  // TODO: Add getConstant* methods for other types.
+
 private:
+  const ASTContext &astContext;
```

Comment:
It does not seem logically correct to require `ASTContext` inside `SpirvContext`. AST is essentially the previous stage before SPIR-V IMR; ideally we should not rely on any AST related stuff at the SPIR-V IMR stage. (We still need QualType, which is kinda unfortunate, but even QualType will hopefully not be used later in SPIR-V IMR transversals.)

Reply:
The only reason I added the ASTContext was to be able to get astContext.UnsignedIntTy as QualType for constructing the SpirvConstantInteger... I'm not sure how to get around that...
In general, I'm not happy with the design / API for creating or getting constants... Also the way we do a linear search to find any existing ones... Not sure what a better way would be.

Do you have any suggestions?